### PR TITLE
Add support for interface

### PR DESCRIPTION
**Pull request by bclermont**

### CODE REVIEW

File: envconfig_test.go [hunk 1]
```diff
@@ -70,6 +75,7 @@ type EmbeddedButIgnored struct {
 
 func TestProcess(t *testing.T) {
 	var s Specification
+	s.NestedInterface = &Anonymous{}
```
Comment (teepark) [inline envconfig_test.go]: Since you have to do this anyway, why wouldn't a user just define the struct field as an `Anonymous`?

Comment (bclermont) [inline envconfig_test.go]: My use case is that I have a single `struct` used by ~20 micro-services:

```
type Configuration struct {
  Logging struct {
    // syslog settings
    // remote logging settings
    // and few more
  }
  Metrics struct {
    // 10 settings about metrics
  }
  GrpcServer struct {
   // TLS settings
   // Port
   // some other grpc specifics
  }
  // and many other settings that all services shared

  Service interface{} // service specifics
}

// and here goes all code that either load file from yaml, json, other source, set development values, etc
```

and in every microservices:

```
type serviceConfiguration struct {
  BusinessLogicServiceSpecific bool
  Other string
}

func main() {
  config := &generic.Configuration{Service: &serviceConfiguration{}}
  config.LoadJSON("/etc/service.json")
}
```

Comment (teepark) [inline envconfig_test.go]: this seems backwards -- you could embed the generic/shared configuration struct in the specific one, then you don't need the `interface{}` placeholder.

Comment (bclermont) [inline envconfig_test.go]: Yeah, I tried to explains the low-level version of my use case. but in real-life my use case is more complicated.

I could only achieve this with an `interface{}`

**Comment by bclermont** [comment]
I will just use keep my fork for me

**Comment by teepark** [comment]
@bclermont please feel free to reopen with clarifications. I don't mean to chase you away, just want to get my head fully wrapped around the need.

**Pull request closed without merging**